flytekit/remote/remote.py

```diff
@@ -790,20 +790,23 @@ def upload_file(
             filename=to_upload.name,
         )
 
+        extra_headers = self.get_extra_headers_for_protocol(upload_location.native_url)
         encoded_md5 = b64encode(md5_bytes)
         with open(str(to_upload), "+rb") as local_file:
             content = local_file.read()
             content_length = len(content)
+            headers = {"Content-Length": str(content_length), "Content-MD5": encoded_md5}
+            headers.update(extra_headers)
             rsp = requests.put(
                 upload_location.signed_url,
                 data=content,
-                headers={"Content-Length": str(content_length), "Content-MD5": encoded_md5},
+                headers=headers,
                 verify=False
                 if self._config.platform.insecure_skip_verify is True
                 else self._config.platform.ca_cert_file_path,
             )
 
-            if rsp.status_code != requests.codes["OK"]:
+            if rsp.status_code not in (requests.codes["OK"], requests.codes["created"]):
```

Comment:
why do we have to check this value as well?

Reply:
Ah sorry, forgot to specify in my PR description. 
Azure returns an HTTP 201. So this was throwing an exception even though the request was successful

Reply:
Done